src/Hosting.CommandLine/Internal/CommandLineService.cs

```diff
@@ -24,23 +23,23 @@ internal class CommandLineService<T> : IDisposable, ICommandLineService where T
         /// <param name="logger">A logger</param>
         /// <param name="state">The command line state</param>
         /// <param name="serviceProvider">The DI service provider</param>
-        public CommandLineService(ILogger<CommandLineService<T>> logger, CommandLineState state,
-            IServiceProvider serviceProvider)
+        /// <param name="configure">The delegate to configure the app</param>
+        public CommandLineService(ILogger<CommandLineService> logger, CommandLineState state,
+            IServiceProvider serviceProvider, Action<CommandLineApplication, IServiceProvider> configure)
         {
             _logger = logger;
             _state = state;
 
-            logger.LogDebug("Constructing CommandLineApplication<{type}> with args [{args}]",
-                typeof(T).FullName, string.Join(",", state.Arguments));
-            _application = new CommandLineApplication<T>(state.Console, state.WorkingDirectory);
-            _application.Conventions
-                .UseDefaultConventions()
-                .UseConstructorInjection(serviceProvider);
```

Comment:
Can you restore this? It should make it possible to eliminate the extra `provider` argument for `configure`.